cpp/src/rolling/jit/kernel.cu

```diff
@@ -66,8 +66,8 @@ __global__ void gpu_rolling_new(cudf::size_type nrows,
     cudf::size_type following_window = get_window(following_window_begin, i);
 
     // compute bounds
-    cudf::size_type start       = min(nrows, max(0, i - preceding_window + 1));
-    cudf::size_type end         = min(nrows, max(0, i + following_window + 1));
+    cudf::size_type start = min(nrows, max(0, static_cast<size_type>(i) - preceding_window + 1));
+    cudf::size_type end   = min(nrows, max(0, static_cast<size_type>(i) + following_window + 1));
```

Comment:
I see that `gpu_rolling` in `cpp/src/rolling/rolling_detail.cuh` has a warning `// to prevent overflow issues when computing bounds use int64_t`, but that's missing in this kernel `gpu_rolling_new`.

What's the relationship here? Does this need to use `int64_t`?

Reply:
Oh, right.  A separate issue from the loop index overflow.  I suppose that's true:  row = (2^31-1), following_window = 2.  I'll fix.